Submitty/Submitty
# [Feature:RainbowGrades] automatic grade summary generation

### Please check if the PR fulfills these requirements:


### What is the current behavior?
Before the web-based rainbow grades ui could generate rainbow grades you had to manually generate grade summaries

### What is the new behavior?
Pressing the save button on the rainbow grades web-based ui form will now automatically generate grade summaries and rainbow grades as long as instructor level credentials have been provided.

### To Test
- First ensure the [jobs daemon is running](https://submitty.org/developer/jobs_daemon) by calling ```sudo systemctl start submitty_daemon_jobs_handler``` prior to running the submitty install script
- Ensure the RainbowGrades repo is cloned in your GIT_CHECKOUT directory
- Modify ```/usr/local/submitty/config/submitty_admin.json``` to provide instructor level credentials, for example instructor/instructor on a regular vagrant development installation
- Log into the 'sample'
- Press the grade reports button
- Press the 'Web-Based Rainbow Grades Generation' button
- Play with the new customization user interface, saving the form when satisfied
- Check out the results in the 'Rainbow Grades' tab (must be enabled inside the course configuration page)

### Other info
- In the event that no credentials or bad credentials are provided the web-ui will still work as normal provided grade summaries have been manually generated.

### Other information?
Tested manually

### Other Other Information?
Credit to @zjxiaohan for his contribution of generate_grade_summaries.py
